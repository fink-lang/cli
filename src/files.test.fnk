{describe, it, expect, to_equal, to_match_snapshot} = import `@fink/jest`

{walk, expand_source, get_files} = import `./files`


describe:: get_files, fn:

  it:: `gets single file`, fn:
    [...files] = get_files:: [`./src/files.test.fnk`], `./build/`

    expect::
      files
      to_equal::
        list: {
          src_path: `src/files.test.fnk`,
          out_path: `build/files.test.js`,
          rel_path: `files.test.fnk`
        }


  it:: `gets single file no putput`, fn:
    [...files] = get_files:: [`./src/files.test.fnk`], null

    expect::
      files
      to_equal::
        list: {
          src_path: `src/files.test.fnk`,
          out_path: null,
          rel_path: `files.test.fnk`
        }


  it:: `ignores files`, fn:
    [...files] = get_files:: [`./src/files.test.fnk`], null, `./src/*.test.fnk`

    expect::
      files
      to_equal:: []


  it:: `gets files from dir`, fn:
    [...files] = get_files:: [`./src`], `./build/test`

    expect::
      files
      to_match_snapshot


