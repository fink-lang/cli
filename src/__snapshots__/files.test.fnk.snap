// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get_files gets files from dir 1`] = `
Array [
  Object {
    "out_path": "build/test/compile.js",
    "rel_path": "compile.fnk",
    "src_path": "src/compile.fnk",
  },
  Object {
    "out_path": "build/test/compile.test.js",
    "rel_path": "compile.test.fnk",
    "src_path": "src/compile.test.fnk",
  },
  Object {
    "out_path": "build/test/files.js",
    "rel_path": "files.fnk",
    "src_path": "src/files.fnk",
  },
  Object {
    "out_path": "build/test/files.test.js",
    "rel_path": "files.test.fnk",
    "src_path": "src/files.test.fnk",
  },
  Object {
    "out_path": "build/test/index.js",
    "rel_path": "index.fnk",
    "src_path": "src/index.fnk",
  },
  Object {
    "out_path": "build/test/index.test.js",
    "rel_path": "index.test.fnk",
    "src_path": "src/index.test.fnk",
  },
  Object {
    "out_path": "build/test/testing.js",
    "rel_path": "testing.fnk",
    "src_path": "src/testing.fnk",
  },
  Object {
    "out_path": "build/test/testing.test.js",
    "rel_path": "testing.test.fnk",
    "src_path": "src/testing.test.fnk",
  },
  Object {
    "out_path": "build/test/yargs.js",
    "rel_path": "yargs.fnk",
    "src_path": "src/yargs.fnk",
  },
]
`;
